### DISCUSSION TITLE
comptime: declare variable from field type number

### DISCUSSION THREAD

**Discussion by Terisback** [vlang/v#11839] [Ideas]
I want to be able to declare a variable from the field data of a structure

### Possible syntax 
```v
obj := $typeof(field.typ){} // where FieldData.typ is int (idx)
```
```v
obj := $(field.typ){} 
```
```v
obj := $(typeof(field.typ).name){}
```

Already found that `table.get_type_symbol(typ)` satisfies the goal

**Comment by Apocryphon-X** [comment]
`v.ast` will not work in my case. Is there another way to do this? 🤔
